flask_tus.py

```diff
@@ -52,76 +91,112 @@ def redis_connection(self):
+        print('begin tus_proprietary_get_file_exists')
         response = make_response("", 200)
+        if 'Upload-Metadata' in request.headers:
+            response.headers['Upload-Metadata'] = request.headers['Upload-Metadata']
+
+        metadata = self._parse_metadata()
+
+        if metadata.get("filename", None) is None:
+            return make_response("metadata filename is not set", 404)
+
+        (filename_name, extension) = os.path.splitext(metadata.get("filename"))
+        if filename_name.upper() in [os.path.splitext(f)[0].upper() for f in
+                                     os.listdir(os.path.dirname(self.upload_folder))]:
+            response.headers['Tus-File-Name'] = metadata.get("filename")
+            response.headers['Tus-File-Exists'] = True
+        else:
+            response.headers['Tus-File-Exists'] = False
+        return response
+
 
-        if request.method == 'GET':
-            metadata = {}
-            for kv in request.headers.get("Upload-Metadata", None).split(","):
-                (key, value) = kv.split(" ")
-                metadata[key] = base64.b64decode(value)
-
-            if metadata.get("filename", None) is None:
-                return make_response("metadata filename is not set", 404)
-
-            (filename_name, extension) = os.path.splitext( metadata.get("filename"))
-            if filename_name.upper() in [os.path.splitext(f)[0].upper() for f in os.listdir( os.path.dirname( self.upload_folder ))]:
-                response.headers['Tus-File-Name'] = metadata.get("filename")
-                response.headers['Tus-File-Exists'] = True
-            else:
-                response.headers['Tus-File-Exists'] = False
+    def create_resource_id(self):
+        return str(uuid.uuid4())
+
+
+    def create_url(self, resource_id):
+        return '{}/{}/{}'.format(request.url_root, self.upload_url, resource_id)
+
+
+    def tus_creation_1_create(self):
+        """Implements POST to create file according to Tus protocol Creation extension"""
+        print('begin tus_creation_1_create')
+        response = make_response("", 200)
+        response.headers['Tus-Resumable'] = self.tus_api_version
+        response.headers['Tus-Version'] = self.tus_api_version_supported
+
+        if 'Upload-Metadata' in request.headers:
+            response.headers['Upload-Metadata'] = request.headers['Upload-Metadata']
+
+        if request.headers.get("Tus-Resumable") is None:
+            self.app.logger.warning("Received File upload for unsupported file transfer protocol")
+            response.data = "Received File upload for unsupported file transfer protocol"
+            response.status_code = 500
             return response
 
-        elif request.method == 'OPTIONS' and request.headers.get('Access-Control-Request-Method', None) is not None:
-            # CORS option request, return 200
+        # process upload metadata
+        metadata = self._parse_metadata()
+
+        if os.path.lexists(
```

Comment:
Guessing this is `os.path.exists` ?

Reply:
Well... I didn't write that line, I just modified it. My personal rule is, don't change something unless I don't fully understand the implications of my change. Given this call works, AFAIK, I didn't change it. I suggest leaving it unless you understand why it is `lexists` and not `exists`.